doc/book/examples/sources/basics/shared-object.move

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// Unlike `Owned` objects, `Shared` ones can be accessed by anyone on the
+/// network. Extended functionality and accessibility of this kind of objects
+/// requires additional effort by securing access if needed.
+module 0x0::shared {
```

Comment:
I think this could be expressed with owned objects by hardcoding the payment address and price.

Perhaps this could be addressed by adding functionality that requires sharing (e.g., a function to change the price or a max supply on the items)? 

I also think this would be a bit more understandable if it were less abstract--e.g., maybe we have a `DonutShop`, items are `Donut`'s. `Shop` and `Item` are generic enough that it makes me expect that we are going to let users list their own items to sell.